### PR TITLE
[manager] Support initialization/allocation of weights 

### PR DESCRIPTION
**Pull request by kparichay**
Support initialization and allocation of weights for LayerV2

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: nntrainer/graph/network_graph.cpp [hunk 1]
```diff
@@ -704,6 +704,7 @@ NetworkGraph::updateRunContext(std::shared_ptr<Manager> &manager,
    */
   // const RunLayerContext &run_context = lnode->getRunContext();
   lnode->updateRunContext(RunLayerContext(
+    // TODO: update weights spec for trainable based on layer trainable prop
```
Comment (zhoonit) [inline nntrainer/graph/network_graph.cpp:707]: Shouldn't this must be done inside getWeightsSpec already before come to this spot.

**Review by zhoonit — APPROVED** [review]
LGTM

File: nntrainer/graph/network_graph.cpp (hunk 1 above)
Comment (kparichay) [reply nntrainer/graph/network_graph.cpp:707]: It depends on when we allow to change trainable property. Maybe this should be done in the end of after finalize() in layer_node.h

Comment (kparichay) [reply nntrainer/graph/network_graph.cpp:707]: I will add its support in #1312 or its next PR.

Comment (zhoonit) [reply nntrainer/graph/network_graph.cpp:707]: Okay

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**